sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala

```diff
@@ -96,9 +96,12 @@ class SQLMetric(
 
   def +=(v: Long): Unit = add(v)
 
-  // _value may be uninitialized, in many cases being -1. We should not expose it to the user
-  // and instead return 0.
-  override def value: Long = if (isZero) 0 else _value
+  // We use -1 as initial value of the SIZE and TIMIMG accumulators (0 is a valid metric value).
+  // We need to return it as it is so that the SQL UI can filter out the invalid accumulator
+  // values in `SQLMetrics.stringValue` when calculating min, max, etc.
+  // However, users accessing the values in the physical plan programmatically still gets -1. They
+  // may use `SQLMetric.isZero` before consuming this value.
+  override def value: Long = _value
```

Comment:
Well, actually, this is a direct reverting of #39311 .
- In this case, we need a review from the author because he proposed this as a kind of regression for accumulator values.
- In addition, could you add a new unit test case for your case? IIUC, this PR only seems to update the existing test cases.

cc @cloud-fan and @viirya and @HyukjinKwon from #39311 , too

Reply:
Added the test

Reply:
`-1` is an internal initial value, and we can't expose it to users. People can get the physical plan and directly access the `SQLMetrics` instances to get the values.